### ISSUE TITLE
upgrade to eks 3.0.1

### ISSUE THREAD

**Issue by jdavredbeard** [pulumi/pulumi-self-hosted-installers#181]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details
In order to keep up with pulumi eks features, upgrade pulumi/eks to 3.0.1

**Comment by jdavredbeard** [comment]
now that https://github.com/pulumi/pulumi-eks/pull/1496 is closed, this ticket is unblocked

**Comment by rshade** [comment]
## 🎯 Full Pulumi Dependencies Upgrade

This issue should be expanded to include a comprehensive upgrade of all core Pulumi dependencies to their latest versions:

### 📦 Target Versions
- **Pulumi Core**: v3.187.0 (from current versions)
- **Pulumi AWS Provider**: v7.3.0 (from current versions)
- **Pulumi EKS Provider**: v4.0.1 (from current v3.0.1 goal)

### 🚀 Implementation Priority

#### **High Priority - Core Infrastructure**
- [ ] Update `pulumi/pulumi` to v3.187.0 across all TypeScript projects
- [ ] Update `@pulumi/aws` to v7.3.0 in all package.json files
- [ ] Update `@pulumi/eks` to v4.0.1 in EKS-related projects
- [ ] Update Go SDK dependencies to match v3.187.0
- [ ] Update `github.com/pulumi/pulumi-aws/sdk/v6` to v7 equivalents

#### **Medium Priority - Platform Installers**
- [ ] **EKS-hosted**: All 8 stages (01-iam through 08-kubernetes-infrastructure)
- [ ] **ECS-hosted**: Both TypeScript and Go implementations
- [ ] **AKS-hosted**: All 3 stages (infrastructure, kubernetes, application)
- [ ] **GKE-hosted**: All 3 stages (infrastructure, kubernetes, application)
- [ ] **Components-microstacks**: API, console, cert-manager, OpenSearch components

#### **Low Priority - Development & Testing**
- [ ] Update quickstart-docker-compose dependencies
- [ ] Update integration test dependencies
- [ ] Update development tooling (documentation generation)

### 📝 Critical Files to Focus On

#### **Package Management**
- `eks-hosted/*/package.json` (8 files)
- `ecs-hosted/ts/*/package.json` (3 files)
- `aks-hosted/*/package.json` (3 files)
- `gke-hosted/*/package.json` (3 files)
- `components-microstacks/package.json`
- `ecs-hosted/go/*/go.mod` (3 files)

#### **Version Compatibility**
- All `Pulumi.yaml` project files for engine version compatibility
- TypeScript `tsconfig.json` files for SDK compatibility
- Go module files for SDK v3 compatibility

### 🧪 Testing & Validation

#### **Build Verification**
- [ ] Run `tsc` in all TypeScript projects
- [ ] Run `go build` in all Go modules
- [ ] Verify no breaking changes in component interfaces

#### **Integration Testing**
- [ ] Deploy quickstart-docker-compose with new versions
- [ ] Run integration tests: `cd quickstart-docker-compose/tests && go test -tags=minio ./...`
- [ ] Verify API endpoints and user creation workflows

#### **Documentation Updates**
- [ ] Update version references in README files
- [ ] Validate mermaid diagrams: `npm run validate:standalone`
- [ ] Update configuration examples with any new required fields

### ⚠️ Breaking Changes to Monitor

#### **EKS v4.0.1 Changes**
- Review EKS provider changelog for breaking changes from v3.x
- Update cluster configurations for new default values
- Check addon and node group configuration changes

#### **AWS v7.3.0 Changes**
- Monitor IAM policy and role configuration changes
- Verify VPC and networking resource compatibility
- Check S3 and RDS configuration updates

#### **Pulumi v3.187.0 Changes**
- Review core engine changes affecting stack operations
- Check for new configuration schema requirements
- Verify automation API compatibility

### 🔄 Phased Rollout Strategy

1. **Phase 1**: Update core dependencies and verify builds
2. **Phase 2**: Update EKS-hosted (highest complexity platform)
3. **Phase 3**: Update remaining cloud platforms (ECS, AKS, GKE)
4. **Phase 4**: Update components and testing infrastructure
5. **Phase 5**: Full integration testing and documentation updates

This comprehensive upgrade will ensure all platform installers benefit from the latest Pulumi features, security updates, and provider capabilities.
